### PR TITLE
feat: add auto import for composables and stores

### PR DESCRIPTION
**Pull request by jbaubree**

### CODE REVIEW

File: src/composables/useDarkMode.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+const toggleDark = useToggle(isDark)
+const preferredDark = usePreferredDark()
+
+export default function useDarkMode() {
+  return {
+    isDark,
+    toggleDark,
+    preferredDark,
+  }
+}
```
[the remark this answers is not in the document]
Comment (jbaubree) [reply src/composables/useDarkMode.ts:12]: vite-auto-import-resolvers works only with export default for the moment :/

[the remark this answers is not in the document]
Comment (jbaubree) [reply src/composables/useDarkMode.ts:12]: We can also do :
```
const isDark = useDark()

export default {
  isDark,
  toggleDark: useToggle(isDark),
  preferredDark: usePreferredDark(),
}
```

and import with `const { isDark, preferredDark } = useDarkMode`

You prefer this syntax?

[the remark this answers is not in the document]
Comment (antfu) [reply src/composables/useDarkMode.ts:12]: I don't like this convention as it requires manual labor and will affects the tree-shaking. This means actually obeying the idea of having auto-imports. I do have a plan to migrate unplugin-auto-import to use https://github.com/unjs/unimport underneath, which powers Nuxt's composables auto import https://v3.nuxtjs.org/guide/directory-structure/composables

[the remark this answers is not in the document]
Comment (jbaubree) [reply src/composables/useDarkMode.ts:12]: Great, so we will wait for this migration that seems to be really good

**Pull request closed without merging**